### PR TITLE
[terra-worklist-data-grid] column header updates

### PR DESCRIPTION
**Pull request by cm9361**
### Summary

**What was changed:**
The Worklist Data Grid was updated to allow consumers to specify the sort indicator and error state for columns.

**Why it was changed:**
Consumers of the Worklist Data Grid need the ability to communicate whether an error occurred in the data displayed to the user as well as the sort order of the data. 

### Testing

This change was tested using:

- [x] WDIO
- [x] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [x] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9046 
UXPLAtFORM-9114

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-framework-docs/CHANGELOG.md [hunk 1]
```diff
@@ -26,6 +26,9 @@
   * Added documentation and tests for `terra-worklist-data-grid`.
   * Added example and test for masked cell feature for `terra-worklist-data-grid`.
 
+* Added
+  * Added example and test for masked cell feature for `terra-worklist-data-grid`.
+
```
Comment (sdadn) [inline packages/terra-framework-docs/CHANGELOG.md:31]: Does this need to be under the `unreleased` section?

Comment (cm9361) [reply packages/terra-framework-docs/CHANGELOG.md:31]: i'll move it.

File: packages/terra-worklist-data-grid/src/ColumnHeaderCell.jsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+   */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }),
+};
+
+const ColumnHeaderCell = (props) => {
+  const {
+    column,
+    width,
+    headerHeight,
+    onColumnSelect,
```
Comment (sdadn) [inline packages/terra-worklist-data-grid/src/ColumnHeaderCell.jsx:44]: What happens when onColumnSelect is undefined? Does it break or throw an error/warning?

File: packages/terra-worklist-data-grid/src/proptypes/columnShape.js [hunk 3]
```diff
@@ -9,10 +14,19 @@ const columnShape = PropTypes.shape({
    * String of text to render within the column header cell.
    */
   displayName: PropTypes.string,
+  /**
+   * A string indicating which sorting indicator should be rendered. If not provided, no sorting indicator will be rendered.
+   * If a `component` value is specified, `sortIndicator` will be ignored. One of `ascending`, `descending`.
+   */
+  sortIndicator: PropTypes.oneOf(Object.values(SortIndicators)),
+  /**
+   * Boolean value indicating whether or not the column has an error in the data
```
Comment (sdadn) [inline packages/terra-worklist-data-grid/src/proptypes/columnShape.js:23]:
```suggestion
   * Boolean value indicating whether or not the column has an error in the data.
```

File: packages/terra-worklist-data-grid/tests/jest/ColumnHeaderCell.test.jsx [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+    expect(columnHeader.props().style.height).toBe('150px');
+
+    const headerContainer = columnHeader.find('.header-container');
+    expect(headerContainer.children()).toHaveLength(1);
+    expect(headerContainer.find('span[role="button"]').text().trim()).toBe('Vitals');
+
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('render column header cell with ascending sort', () => {
```
Comment (sdadn) [inline packages/terra-worklist-data-grid/tests/jest/ColumnHeaderCell.test.jsx:41]: nit: grammar
```suggestion
  it('renders a column header cell with ascending sort', () => {
```

File: packages/terra-worklist-data-grid/src/ColumnHeaderCell.jsx (hunk 2 above)
Comment (cm9361) [reply packages/terra-worklist-data-grid/src/ColumnHeaderCell.jsx:44]: Nothing.  This is not required to be passed by consumers.  Consumers would only pass if they want to handle the action.

File: packages/terra-worklist-data-grid/CHANGELOG.md [hunk 5]
```diff
@@ -8,6 +8,7 @@
   * Added masked cell support.
   * Added hover state for selectable cells.
   * Added blank cell support.
+  * Added ability to allow consumers to specify the sort indicator and error state for columns
```
Comment (sdadn) [inline packages/terra-worklist-data-grid/CHANGELOG.md:11]:
```suggestion
  * Added ability to allow consumers to specify the sort indicator and error state for columns.
```

**Review by sdadn — APPROVED** [review]

**Review by eawww — APPROVED** [review]
Announcement works in VO + Safari and JAWS + Edge. 
Keyboard accessible sorting functionality and additional indications for screen readers will be accomplished in another story.

File: packages/terra-worklist-data-grid/src/ColumnHeaderCell.jsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  return (
+  /* eslint-disable react/forbid-dom-props */
+    <th
+      key={column.id}
+      className={cx('column-header', theme.className, { selectable: !(column.isSelectable === false) })}
+      tabIndex="-1"
+      role="columnheader"
+      scope="col"
+      aria-sort={column.sortIndicator}
+      {...(!(column.isSelectable === false) && onColumnSelect && { onClick: () => { onColumnSelect(column.id); } })}
```
Comment (sycombs) [inline packages/terra-worklist-data-grid/src/ColumnHeaderCell.jsx:75]: Is there a difference between doing this and doing something like:
```suggestion
      onClick={(column.isSelectable && onColumnSelect) ? handleColumnSelect : undefined }
```

And then define `handleColumnSelect` to be something like this in terra-clinical-data-grid: https://github.com/cerner/terra-clinical/blob/main/packages/terra-clinical-data-grid/src/subcomponents/Cell.jsx#L100-L106

Where onClick is defined as: https://github.com/cerner/terra-clinical/blob/main/packages/terra-clinical-data-grid/src/subcomponents/Cell.jsx#L167

**Comment by mjpalazzo** [comment]
Great work on the sort indicator and error state, @cm9361 and @sdadn! It is so exciting to see the evolution of the grid functionality and team work!

**Comment by mjpalazzo** [comment]
Validated with Edge+JAWS and Safari+VO.
Sort functionality works with mouse. Is sort with keyboard coming next?
JAWS and VO announced error indicator.

**Comment by cm9361** [comment]
> Validated with Edge+JAWS and Safari+VO. Sort functionality works with mouse. Is sort with keyboard coming next? JAWS and VO announced error indicator.

Yes, the keyboard will be coming next along with the other keyboard selection features.

**Pull request merged**